irohad/simulator/impl/simulator.cpp

```diff
@@ -66,17 +66,25 @@ namespace iroha {
         const shared_model::interface::Proposal &proposal) {
       log_->info("process proposal");
       // Get last block from local ledger
-      block_queries_->getTopBlocks(1)
-          .subscribe_on(rxcpp::observe_on_new_thread())
-          .as_blocking()
-          .subscribe([this](auto block) { last_block = block; });
-      if (not last_block) {
-        log_->warn("Could not fetch last block");
+      auto block_fetched = false;
```

Comment:
You can use the same scheme as with block_loader_impl to avoid extra bool. It is still not very pretty, but we need some additional refactoring to make it prettier, so this is out of scope of this pr

Reply:
To perform he same schema here, I should nullify a class variable last_block, which does not sound very good... another variant is to place the rest of function under lambda "good" branch, but this is even worse

Reply:
I see, then we can return boolean from match and initialize block_fetched there